internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go

```diff
@@ -24,66 +21,108 @@ func counterExtractor(m *dto.Metric) float64 {
 	return m.GetCounter().GetValue()
 }
 
-func convertMetricFamily(mf *dto.MetricFamily) []*datapoint.Datapoint {
+func convertMetricFamily(sm pmetric.ScopeMetrics, mf *dto.MetricFamily) {
 	if mf.Type == nil || mf.Name == nil {
-		return nil
+		return
 	}
 	switch *mf.Type {
 	case dto.MetricType_GAUGE:
-		return makeSimpleDatapoints(*mf.Name, mf.Metric, sfxclient.GaugeF, gaugeExtractor)
+		makeGaugeDataPoints(sm, *mf.Name, mf.Metric, gaugeExtractor)
 	case dto.MetricType_COUNTER:
-		return makeSimpleDatapoints(*mf.Name, mf.Metric, sfxclient.CumulativeF, counterExtractor)
+		makeSumDataPoints(sm, *mf.Name, mf.Metric, counterExtractor)
 	case dto.MetricType_UNTYPED:
-		return makeSimpleDatapoints(*mf.Name, mf.Metric, sfxclient.GaugeF, untypedExtractor)
+		makeGaugeDataPoints(sm, *mf.Name, mf.Metric, untypedExtractor)
 	case dto.MetricType_SUMMARY:
-		return makeSummaryDatapoints(*mf.Name, mf.Metric)
+		makeSummaryDatapoints(sm, *mf.Name, mf.Metric)
 	// TODO: figure out how to best convert histograms, in particular the
 	// upper bound value
 	case dto.MetricType_HISTOGRAM:
-		return makeHistogramDatapoints(*mf.Name, mf.Metric)
+		makeHistogramDatapoints(sm, *mf.Name, mf.Metric)
 	default:
-		return nil
 	}
 }
 
-func makeSimpleDatapoints(name string, ms []*dto.Metric, dpf dpFactory, e extractor) []*datapoint.Datapoint {
-	dps := make([]*datapoint.Datapoint, len(ms))
-	for i, m := range ms {
-		dps[i] = dpf(name, labelsToDims(m.Label), e(m))
+func makeGaugeDataPoints(sm pmetric.ScopeMetrics, name string, ms []*dto.Metric, e extractor) {
+	metric := sm.Metrics().AppendEmpty()
+	metric.SetName(name)
+	g := metric.SetEmptyGauge()
+	for _, m := range ms {
+		dp := g.DataPoints().AppendEmpty()
+		dp.SetDoubleValue(e(m))
+		for i := range m.Label {
+			dp.Attributes().PutStr(m.Label[i].GetName(), m.Label[i].GetValue())
+		}
+		dp.Attributes().PutStr("system.type", "prometheus-exporter")
 	}
-	return dps
 }
 
-func makeSummaryDatapoints(name string, ms []*dto.Metric) []*datapoint.Datapoint {
-	var dps []*datapoint.Datapoint
+func makeSumDataPoints(sm pmetric.ScopeMetrics, name string, ms []*dto.Metric, e extractor) {
+	metric := sm.Metrics().AppendEmpty()
+	metric.SetName(name)
+	sum := metric.SetEmptySum()
+	sum.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+	sum.SetIsMonotonic(true)
+	for _, m := range ms {
+		dp := sum.DataPoints().AppendEmpty()
+		dp.SetDoubleValue(e(m))
+		for i := range m.Label {
+			dp.Attributes().PutStr(m.Label[i].GetName(), m.Label[i].GetValue())
+		}
+		dp.Attributes().PutStr("system.type", "prometheus-exporter")
+	}
+}
+
+func makeSummaryDatapoints(sm pmetric.ScopeMetrics, name string, ms []*dto.Metric) {
```

Comment:
I'm not sure these or histograms have existing test coverage. Updating the [prometheus-exporter integration test](https://github.com/signalfx/splunk-otel-collector/tree/main/tests/receivers/smartagent/prometheus-exporter) to include them (and exercise both suggested paths) would be helpful in vetting the changes.

Reply:
Yes, I'll create a separate PR to add an integration test with histograms, and then rebase this PR to use the same integration test and ensure backwards compatibility.